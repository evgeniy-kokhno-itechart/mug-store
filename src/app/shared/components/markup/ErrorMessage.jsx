import React from 'react';
import { PropTypes } from 'prop-types';

const ErrorMessage = ({ message, customClasses }) => (
  <span className={`text-danger ${customClasses}`}>
    {message}
  </span>
);

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  customClasses: PropTypes.string,
};

ErrorMessage.defaultProps = {
  customClasses: '',
};

export default ErrorMessage;
