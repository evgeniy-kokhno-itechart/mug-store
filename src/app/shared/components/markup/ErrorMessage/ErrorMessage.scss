@import 'Variables';

.error-message {
    color: $color-danger;
}