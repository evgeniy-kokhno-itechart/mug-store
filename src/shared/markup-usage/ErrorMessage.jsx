import React from 'react';
import { PropTypes } from 'prop-types';

const ErrorMessage = ({ message }) => <span className="text-danger">{`The following error occurred: ${message}`}</span>;

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
};

export default ErrorMessage;
