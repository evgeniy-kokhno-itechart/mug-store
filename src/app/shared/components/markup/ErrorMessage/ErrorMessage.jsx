import React from 'react';
import { PropTypes } from 'prop-types';
import './ErrorMessage.scss';

const ErrorMessage = ({ message, customClasses }) => (
  <p className={`error-message ${customClasses}`}>
    {message}
  </p>
);

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  customClasses: PropTypes.string,
};

ErrorMessage.defaultProps = {
  customClasses: '',
};

export default ErrorMessage;
