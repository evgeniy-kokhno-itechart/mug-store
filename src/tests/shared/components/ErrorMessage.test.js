/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ErrorMessage from '../../../app/shared/components/markup/ErrorMessage';

configure({ adapter: new Adapter() });

describe('<ErrorMessage />', () => {
  let ErrorMessageWrapper;

  beforeEach(() => {
    ErrorMessageWrapper = shallow(<ErrorMessage message="test error" />);
  });

  it('check ErrorMessage renders properly with default props', () => {
    expect(ErrorMessageWrapper).toMatchSnapshot();
  });

  it('check ErrorMessage renders properly custom class testclass', () => {
    ErrorMessageWrapper.setProps({ customClasses: 'testclass' });
    expect(ErrorMessageWrapper).toMatchSnapshot();
  });
});
