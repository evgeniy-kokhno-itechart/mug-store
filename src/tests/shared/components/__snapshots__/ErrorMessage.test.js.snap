// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> check ErrorMessage renders properly custom class testclass 1`] = `
<p
  className="error-message testclass"
>
  test error
</p>
`;

exports[`<ErrorMessage /> check ErrorMessage renders properly with default props 1`] = `
<p
  className="error-message "
>
  test error
</p>
`;
